contracts/polygon/connectors/hop/main.sol

```diff
@@ -37,24 +37,26 @@ abstract contract Resolver is Helpers {
 		}
 
 		params.amount = getUint(getId, params.amount);
+		TokenInterface tokenContract = TokenInterface(params.token);
 
-		bool isMatic = params.token == maticAddr;
-		params.token = params.token == maticAddr ? wmaticAddr : params.token;
+		if (params.token == wmaticAddr) {
+			convertWmaticToMatic(true, tokenContract, params.amount);
+			params.token = maticAddr;
+		}
 
-		TokenInterface tokenContract = TokenInterface(params.token);
+		bool isNative = params.token == maticAddr;
```

Comment:
@pradyuman-verma We need to check if the token is wnative, then we need to unwrap it.

Reply:
Doing that part in line 42-45